### ISSUE TITLE
crop and mask function not able to write raster

### ISSUE THREAD

**Issue by sm2607** [rspatial/terra#857]
Hi I am using terra package as its working faster compared to raster, but I am stuck at one part, I am trying to mask a raster with another. below is my code.

`masked<-terra::crop(raster,poly_f,mask=TRUE, filename=paste0(path,"asd3.tif"),verbose=T)`

its started creating file with "asd.tif" but its "0 bytes", below are verbose details:-

```
driver        : GTiff
disk available: 72.7 GB
disk needed   : 8 GB
memory avail. : 3.12 GB
memory allow. : 1.87 GB
memory needed : 32.009 GB  (2 copies)
in memory     : false
block size    : 2316 rows
n blocks      : 29
pb            : 1

**Comment by rhijmans** [comment]
This works for me:

```
library(terra)
#terra 1.6.33

r <- rast(system.file("ex/elev.tif", package="terra"))
v <- vect(system.file("ex/lux.shp", package="terra"))[1]

masked <- crop(r, v, mask=TRUE, filename=("test.tif"), overwrite=TRUE)
masked
#class       : SpatRaster 
#dimensions  : 28, 40, 1  (nrow, ncol, nlyr)
#resolution  : 0.008333333, 0.008333333  (x, y)
#extent      : 5.825, 6.158333, 49.95, 50.18333  (xmin, xmax, ymin, ymax)
#coord. ref. : lon/lat WGS 84 (EPSG:4326) 
#source      : test.tif 
#name        : elevation 
#min value   :       335 
#max value   :       547 
```

Perhaps because I am using the development version. You do not mention the version you are using; and your example is not reproducible, so there is not much more that I can say. Can you test with the development version?

**Comment by rhijmans** [comment]
As I have not heard back, I assume this works for you with the dev version.

**Comment by sm2607** [comment]
hi its working for me on windows but getting same issue in linux,
 also I have noticed there is significant decrease in memory available and memory allowed after each raster process. There are almost 32 Gb ram available , still its showing memory available 0.5 Gb and Memory allowed 0.3 Gb,  in earlier step it was 18 and 12 respectively. is there any way to increase this, and prevent it to decrease after each step . I tried gc(), after each step still no improvement.
